R/scripts/create_seeding.R

```diff
@@ -38,61 +38,126 @@ library(tidyr)
+  optparse::make_option(c("-c", "--config"), action="store", default=Sys.getenv("CONFIG_PATH"), type='character', help="path to the config file"),
+  optparse::make_option(c("-s", "--source"), action="store", default="CSSE", type='character', help="source of case data: USAFacts or CSSE"),
+  optparse::make_option(c("-d", "--data"), action="store", default=file.path("data","case_data","case_data.csv"), type='character', help="path to the case data file")
 )
 
-opts = optparse::parse_args(optparse::OptionParser(option_list=option_list))
+opt = optparse::parse_args(optparse::OptionParser(option_list=option_list))
 
-config <- covidcommon::load_config(opts$c)
+config <- covidcommon::load_config(opt$c)
 if (length(config) == 0) {
   stop("no configuration found -- please set CONFIG_PATH environment variable or use the -c command flag")
 }
 
-jhucsse <- covidImportation::get_clean_JHUCSSE_data(aggr_level = "UID", 
-                                   last_date = as.POSIXct(lubridate::ymd(config$end_date)),
-                                   case_data_dir = file.path('importation',config$spatial_setup$setup_name,"case_data"),
-                                   save_raw_data=TRUE,
-                                   us_data_only=FALSE)
-
-print("Successfully pulled JHU CSSE data for seeding.")
-
 all_times <- lubridate::ymd(config$start_date) +
   seq_len(lubridate::ymd(config$end_date) - lubridate::ymd(config$start_date))
 
-geodata <- report.generation:::load_geodata_file(file.path(config$spatial_setup$base_path, config$spatial_setup$geodata),5,'0',TRUE)
 
-all_geoids <- geodata[[config$spatial_setup$nodenames]]
 
-incident_cases <- jhucsse %>%
-  dplyr::filter(FIPS %in% all_geoids) %>%
-  dplyr::select(Update, FIPS, incidI)
 
-incident_cases$Update <- as.Date(incident_cases$Update)
+# Get the data
+
+
+
+# get data if a US model  ---------------------------------
+
+if (is.null(config$spatial_setup$us_model) || config$spatial_setup$us_model==TRUE){
+  
+  # Load either JHUCSSE or USAFacts data
+  if (tolower(opt$source) %in% c("csse", "jhucsse", "jhu csse")){
+    jhucsse <- covidImportation::get_clean_JHUCSSE_data(aggr_level = "UID",
+                                     last_date = as.POSIXct(lubridate::ymd(config$end_date)),
+                                     case_data_dir = file.path('importation',config$spatial_setup$setup_name,"case_data"),
+                                     save_raw_data=TRUE,
+                                     us_data_only=FALSE)
+    print("Successfully pulled JHU CSSE data for seeding.")
+    
+  } else if (tolower(opt$source) %in% c("usafacts", "usa facts", "usa")){
+    cases_deaths <- covidcommon::get_USAFacts_data()
+    print("Successfully pulled USAFacts data for seeding.")
+  }
+  
+  geodata <- report.generation:::load_geodata_file(file.path(config$spatial_setup$base_path, config$spatial_setup$geodata),5,'0',TRUE)
```

Comment:
We should either export load_geodata_file, or access via read_file_of_type here

Reply:
which do you prefer? I will export load_geodata_file anyways. but could still do the other option in the script

Reply:
`load_geodata_file` has `geoid` as a required column, and allows padding of those and some standardization.  If you're using those options, and do require the geoid column, then i think that using `load_geodata_file` is preferred.  Otherwise, probably switch to `read_file_of_type`

Reply:
load_geodata_file will be used then